SpatialGDK/Source/SpatialGDK/Public/Schema/ClientRPCEndpoint.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved	
+
+#pragma once	
+
+#include "Schema/Component.h"	
+#include "SpatialConstants.h"	
+#include "Utils/SchemaUtils.h"	
+
+#include <WorkerSDK/improbable/c_schema.h>	
+#include <WorkerSDK/improbable/c_worker.h>	
+
+namespace SpatialGDK
+{
+
+struct ClientRPCEndpoint : Component
+{
+	static const Worker_ComponentId ComponentId = SpatialConstants::CLIENT_RPC_ENDPOINT_COMPONENT_ID;
+
+	ClientRPCEndpoint() = default;
+
+	Worker_ComponentData CreateRPCEndpointData()
+	{
+		Worker_ComponentData Data = {};
+		Data.component_id = ComponentId;
+		Data.schema_type = Schema_CreateComponentData(ComponentId);
+		Schema_Object* ComponentObject = Schema_GetComponentDataFields(Data.schema_type);
+		Schema_AddBool(ComponentObject, SpatialConstants::UNREAL_RPC_ENDPOINT_READY_ID, ready);
+
+		return Data;
+	}
+
+	Worker_ComponentUpdate CreateRPCEndpointUpdate()
+	{
+		Worker_ComponentUpdate Update = {};
+		Update.component_id = ComponentId;
+		Update.schema_type = Schema_CreateComponentUpdate(ComponentId);
+		Schema_Object* UpdateObject = Schema_GetComponentUpdateFields(Update.schema_type);
+		Schema_AddBool(UpdateObject, SpatialConstants::UNREAL_RPC_ENDPOINT_READY_ID, ready);
+
+		return Update;
+	}
+
+	bool ready = false;
```

Comment:
If we are doing unreal style then this should be bReady.

Reply:
However as this is our own type I'd rather make it `Ready` for a public field. But it looks like this is using unreal style everywhere else.

Reply:
I'll change it to Ready.

Reply:
We adhere to the Unreal style guide, so all bools should be prefixed with `b`. The only exception should be when adapting existing engine code where the convention isn't used.

Reply:
Changed to bReady.